letsencrypt-nginx/letsencrypt_nginx/configurator.py

```diff
@@ -366,6 +366,26 @@ def enhance(self, domain, enhancement, options=None):
         except errors.PluginError:
             logger.warn("Failed %s for %s", enhancement, domain)
 
+    def _enable_redirect(self, vhost, unused_options):
+        """Redirect all equivalent HTTP traffic to ssl_vhost.
+
+        Add rewrite directive to non https traffic
+
+        .. note:: This function saves the configuration
+
+        :param vhost: Destination of traffic, an ssl enabled vhost
+        :type vhost: :class:`~letsencrypt_nginx.obj.VirtualHost`
+
+        :param unused_options: Not currently used
+        :type unused_options: Not Available
+        """
+        redirect_block = [[['if', '($scheme != "https")'],
+            [['return', '301 https://$host$uri']]
```

Comment:
Any reason you went with $uri rather than $request_uri? $uri is normalized, changing `/foo/../a` to `/a` (which doesn't matter so much, since browsers do that themselves) but also having some [vaguely documented differences around index files](http://nginx.org/en/docs/http/ngx_http_core_module.html#var_uri) (which might matter, but I can't find a case where it does). My leaning is that $request_uri will have fewer surprises in store for us. (Their additional difference with regard to internal redirects shouldn't matter to us.)